arch/xtensa/core/xtensa-asm2-util.S

```diff
@@ -153,6 +153,10 @@ _restore_context:
 	l32i a0, a1, BSA_PS_OFF
 	wsr a0, ZSR_EPS
 
+#if XCHAL_HAVE_FP && defined(CONFIG_CPU_HAS_FPU) && defined(CONFIG_FPU_SHARING)
```

Comment:
Better to use just one symbol here and not double-test.  Just use a BUILD_ASSERT() to verify that the kconfig and the core-isa.h symbol match.

Reply:
@andyross , thanks for the nitpicks. I'm having some trouble using the BUILD_ASSERT() macro on the xtensa-asm2-util.S or xtensa-asm2-s.h. 

```
#if defined(CONFIG_FPU_SHARING)
BUILD_ASSERT(XCHAL_HAVE_FP & CONFIG_CPU_HAS_FPU)
#endif
```
It keeps failing to compile with the error:
```
zephyr/arch/xtensa/include/xtensa-asm2-s.h: Assembler messages:
zephyr/arch/xtensa/include/xtensa-asm2-s.h:13: Error: unknown opcode or format name 'build_assert'
```

Any tip on using it?

Reply:
Ahh, thanks @dcpleung !